ambry-network/src/main/java/com.github.ambry.network/NetworkClient.java

```diff
@@ -191,17 +215,39 @@ public void close() {
   }
 
   /**
-   * A class that consists of a {@link RequestInfo} with the time at which it is added to the queue.
+   * A class that consists of a {@link RequestInfo} and some metadata related to the request
    */
   private class RequestMetadata {
-    // the time at which this request was queued.
-    long requestQueuedTimeMs;
+    // to track network request related metrics
+    ClientNetworkRequestMetrics clientNetworkRequestMetrics;
     // the RequestInfo associated with the request.
     RequestInfo requestInfo;
+    // the time at which this request was queued.
+    private long requestQueuedTimeMs;
```

Comment:
I know this is old code, but it was a bit confusing as this seemed like elapsed time. Do you think you could call this `requeustQueuedAtMs` and `requestDequeuedAtMs`?